### DISCUSSION TITLE
RetrievalQA Error

### DISCUSSION THREAD

**Discussion by konstantin-kostadinov** [qdrant/qdrant#2987] [Q&A]
When I am using qdrant on docker with RetrievalQA I get this error:
**gRPC /qdrant.Points/Search failed with Client specified an invalid argument "Wrong input: Default vector params are not specified in config"**

What is this default config and where to put it?

**Comment by timvisee** [comment]
Hi @konstantin-kostadinov.

Would you mind to share the snippet of code (or similar) you use to interface with Qdrant? I'm not familiar with RetrievalQA myself.

This error means that you're trying to use vectors in a collection, which are not configured in that collection. Maybe you still need to set up the collection properly, before you can use it to store data?

To be more specific, the exact error is coming from here: https://github.com/qdrant/qdrant/blob/5c36256caf94f7a90e31248415fd643818a09543/lib/collection/src/config.rs#L188-L198

**Comment by konstantin-kostadinov** [comment]
Hi @timvisee 
Thank you for the response.

**Here I load the data:**

```python
from qdrant_client.http import models as rest

vector_size = 1536
chunk_size = 1000
embeddings_url = "https://cdn.openai.com/API/examples/data/vector_database_wikipedia_articles_embedded.zip"

wget.download(embeddings_url)

with zipfile.ZipFile("vector_database_wikipedia_articles_embedded.zip", "r") as zip_ref:
    zip_ref.extractall("../data")


client = QdrantClient(
    host="localhost",
    port=6333,
    prefer_grpc=True,
)

client.recreate_collection(
    collection_name="Articles",
    vectors_config={
        "title": rest.VectorParams(
            distance=rest.Distance.COSINE,
            size=vector_size,
        ),
        "content": rest.VectorParams(
            distance=rest.Distance.COSINE,
            size=vector_size,
        ),
    }
)

for chunk_df in pd.read_csv('../data/vector_database_wikipedia_articles_embedded.csv', chunksize=chunk_size):
    chunk_df["title_vector"] = chunk_df.title_vector.apply(literal_eval)
    chunk_df["content_vector"] = chunk_df.content_vector.apply(literal_eval)
    chunk_df.head()
    points = [
        rest.PointStruct(
            id=k,
            vector={
                "title": v["title_vector"],
                "content": v["content_vector"],
            },
            payload=v.to_dict(),
        )
        for k, v in chunk_df.iterrows()
    ]

    client.upsert(collection_name="Articles", points=points)

articles_count = client.count(collection_name="Articles")
print("{} articles".format(articles_count))
```

**This is the part where the code fails.**

```python
from langchain.chains import RetrievalQA
from langchain.embeddings import OpenAIEmbeddings
from langchain.llms.openai import OpenAI
from langchain.prompts import PromptTemplate
from langchain.vectorstores import Qdrant
from qdrant_client import QdrantClient

collection_name = "Articles"

client = QdrantClient(
    host="localhost",
    port=6333,
    prefer_grpc=True,
)


llm = OpenAI(openai_api_key=OPENAI_API_KEY)
embeddings = OpenAIEmbeddings(openai_api_key=OPENAI_API_KEY)
doc_store = Qdrant(client, collection_name, embeddings)

custom_prompt = """
Use the following pieces of context to answer the question at the end. Please provide
a short single-sentence summary answer only. If you don't know the answer or if it's 
not present in given context, don't try to make up an answer, but suggest me a random 
unrelated song title I could listen to. 
Context: {context}
Question: {question}
Helpful Answer:
"""

custom_prompt_template = PromptTemplate(
    template=custom_prompt, input_variables=["context", "question"]
)

qa = RetrievalQA.from_chain_type(
    llm=llm,
    chain_type="stuff",
    retriever=doc_store.as_retriever(),
    chain_type_kwargs={"prompt": custom_prompt_template},
    return_source_documents=False,
)

answer = qa.run(question)
print(answer)

```

**Reply by timvisee** [reply]
You create a collection with `title` and `content` vectors, which seems fine. However, in your second snippet it searches through Qdrant with unnamed vectors. That's why the above error comes up.

It is unclear to me however how to tweak the search step to search for `title` and `content` vectors instead. I've read through the [RetrievalQA docs](https://js.langchain.com/docs/modules/chains/popular/vector_db_qa) but couldn't find the answer. I'm unfamiliar with it.

Would it be possible to embed both the title and content together into a single unnamed vector, or does that not work for your use case? If that's possible, you likely don't have to modify your second snippet.

Bu the way, please be aware that you use `recreate_collection`, which will delete any current collection every time you run it. You may consider to use `create_collection` instead.
